### ISSUE TITLE
Sonatype PublishSigned support?

### ISSUE THREAD

**Issue by lihaoyi** [com-lihaoyi/mill#5]
One of the major things Mill is currently missing is the ability to publishSigned artifacts (along with source jars, etc.) to Maven central.

We already have the ability to create Jars, e.g. `mill run ScalaPlugin.assembly`. https://github.com/lihaoyi/mill/issues/3 will allow us to make them executable. The last step is to let us publish them to some artifact repository: Maven Central, Bintray, etc.

I'm not familiar with what would go into uploading a Jar to maven central, but some good resources may be:

- https://stackoverflow.com/questions/28846802/how-to-manually-publish-jar-to-maven-central
- SBT PGP source code https://github.com/sbt/sbt-pgp, since that SBT plugin seems able to do this

Basically whatever SBT-PGP is doing, we should be able to do the same
